src/Illuminate/Database/Query/Grammars/Grammar.php

```diff
@@ -177,7 +177,19 @@ protected function compileJoinConstraint(array $clause)
     {
         $first = $this->wrap($clause['first']);
 
-        $second = $clause['where'] ? '?' : $this->wrap($clause['second']);
+        if ($clause['where']) {
+
```

Comment:
This were done, for readability, I mean
```php
        if ($clause['where']) {

            if ($clause['operator'] === 'in' || $clause['operator'] === 'not in') {
                // In case of `whereIn` operator.
                $second = '(' . join(', ', array_fill(0, $clause['second'], '?')) . ')';
            } else {
                // In case of common `where` operator.
                $second = '?';
            }

        } else {
            $second = $this->wrap($clause['second']);
        }
```

looks better, than

```php
        if ($clause['where']) {
            if ($clause['operator'] === 'in' || $clause['operator'] === 'not in') {
                // In case of `whereIn` operator.
                $second = '(' . join(', ', array_fill(0, $clause['second'], '?')) . ')';
            } else {
                // In case of common `where` operator.
                $second = '?';
            }
        } else {
            $second = $this->wrap($clause['second']);
        }
```

Am I wrong?